### DISCUSSION TITLE
generate entities without prelude [sea-orm-cli]

### DISCUSSION THREAD

**Discussion by hahouari** [SeaQL/sea-orm#2283] [Ideas]
So essentially I'm currently running `sea-orm-cli generate entity -o api/src/entities --with-serde=both` to generate Rust structs for my tables, my problem is it generates an unnecessary file `prelude.rs` that contains the following
```rust
//! `SeaORM` Entity, @generated by sea-orm-codegen 1.0.0-rc.7

pub use super::tiers::Entity as Tiers;
pub use super::users::Entity as Users;
```

I don't need the file since I use for instance `users::Entity` as it better works with my VS Code settings and import automatically, while prelude aliases don't do that, so I don't use them anyway, and the unused prelude import flooded my terminal with warnings:

![image](https://github.com/SeaQL/sea-orm/assets/39862612/2f4b49e2-3ad2-4045-9d74-03b9b3888af3)

I double checked latest version docs, GitHub discussions/issues, and there is nothing on how to skip generating such file.

I have 2 suggestions here:
1. add the option to skip generating the file through cli.
2. Even better, add option to append the prelude file with this trait header `#![allow(unused)]` on its top, that way these preludes are there for future use, and no warnings on not using them, in fact I believe this option should be default, but you might know better on this.

At the end, I place my special thanks to you guys for such automation tool, it really saved me hours of my days.

### Update

I created a PR https://github.com/SeaQL/sea-orm/pull/2322 that allows customizing the `prelude.rs` generation through CLI arguments.

**Comment by musjj** [comment]
This has also been bothering me. Why not just generate a type alias in the entity file itself? Like this:

```rust
type User = Entity;
```

Auto-generated preludes are awkward, because the usual (unspoken?) convention is to place a single prelude on the crate level. So if your entity module lives in the same crate the rest your project is in, it kinda sticks out like a sore thumb. Worst if your project is not using the `prelude` convention in the first place.

**Reply by hahouari** [reply]
I made a PR to fix this, but it is yet to be reviewed and merged...

https://github.com/SeaQL/sea-orm/pull/2322

**Reply by musjj** [reply]
Thanks, hope that will get merged. For now I added this to my `justfile` script:

```bash
sed -e 1d -e "s/^pub use super::/pub use /" prelude.rs >>mod.rs
sed -i -z 's/\npub mod prelude\;\n//' mod.rs
rm prelude.rs
```

You still get the nice alias for entities, but now they're placed inside `mod.rs` instead of `prelude.rs`

**Comment by Razzwan** [comment]
Greetings! Thank you for the great ORM for Rust that you are creating (so far I only dream of contributing). 

Completely agree, that this cli option is useful

If you agree that this will be a good improvement, then may I help somehow?
